src/eduid_userdb/support/models.py

```diff
@@ -44,48 +39,29 @@ def __init__(self, data):
             self.update(_data)
 
 
-class SupportUser(GenericFilterDict):
-
-    remove_keys = ['_id', 'letter_proofing_data']
-
-    def __init__(self, data):
-        _data_in = User(data).to_dict()
-        _user_id = _data_in['_id']
-        super(SupportUser, self).__init__(_data_in)
-
-        self['user_id'] = _user_id
-        self['mailAliases'] = [MailAlias(alias) for alias in self['mailAliases']]
-        self['passwords'] = [Password(password) for password in self['passwords']]
-        self['tou'] = [ToU(tou) for tou in self['tou']]
-
-
-class SupportDashboardUser(GenericFilterDict):
+class SupportUserFilter(GenericFilterDict):
 
     remove_keys = ['_id', 'letter_proofing_data']
 
     def __init__(self, data):
-        _data_in = DashboardUser(data=data).to_dict()
-        _user_id = _data_in['_id']
-        super(SupportDashboardUser, self).__init__(_data_in)
+        _data = deepcopy(data)
+        super(SupportUserFilter, self).__init__(_data)
 
-        self['user_id'] = _user_id
         self['mailAliases'] = [MailAlias(alias) for alias in self['mailAliases']]
-        self['passwords'] = [Password(password) for password in self['passwords']]
+        self['passwords'] = [Credential(password) for password in self['passwords']]
```

Comment:
I think you should let userdb parse data into a User() of some sort and then build your dict from the User object. The dict representations of users is subject to change.

Reply:
The in data comes from User().to_dict. The GenericFilterDict wont fail on missing data so if the underlying user dict changes we have to update our models but it won't crash.